File: 1568-pseudo-palindromic-paths-in-a-binary-tree/pseudo-palindromic-paths-in-a-binary-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
int findPath(struct TreeNode* root,int* hash){
    hash[root->val]++;
    if(root->left == NULL && root->right == NULL){
        int count = 0;
        for(int i = 1;i <= 9;i++){
            if(hash[i] % 2 == 1){
                count++;
            }
        }
        hash[root->val]--;
        if(count>1){
            return 0;
        }
        else{
            return 1;
        }
    }
    int re = 0;
    if(root->left != NULL){
        re += findPath(root->left,hash);
    }
    if(root->right != NULL){
        re += findPath(root->right,hash);
    }
    hash[root->val]--;
    return re;
}

int pseudoPalindromicPaths (struct TreeNode* root) {
    int* hash = (int*)calloc(10,sizeof(int));
    return findPath(root,hash);
}

File: 1568-pseudo-palindromic-paths-in-a-binary-tree/pseudo-palindromic-paths-in-a-binary-tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public int PseudoPalindromicPaths (TreeNode root) {
        return CountPseudoPalindromicPaths(root, 0);
    }

    private int CountPseudoPalindromicPaths(TreeNode node, int count) {
        if (node == null) {
            return 0;
        }

        count ^= 1 << (node.val - 1);

        if (node.left == null && node.right == null) {
            // Check if the current path is pseudo-palindromic
            return (count & (count - 1)) == 0 ? 1 : 0;
        }

        int leftPaths = CountPseudoPalindromicPaths(node.left, count);
        int rightPaths = CountPseudoPalindromicPaths(node.right, count);

        return leftPaths + rightPaths;
    }
}